modules/Bio/EnsEMBL/Compara/GeneTreeNode.pm

```diff
@@ -306,17 +307,37 @@ sub root {
 =cut
 
 sub is_leaf {
-  my $self = shift;
-
+    my $self = shift;
     my $child_count = $self->get_child_count;
-    if ( $child_count == 0 ) {
-        return 1;
-    } elsif ( $child_count == 1 && $self->tree->tree_type eq 'supertree' ) {
+    return 1 if ($child_count == 0);
+    if ( $self->is_supertree and $child_count == 1 ) {
         my $child = $self->children->[0];
-        return ($child->node_id == $child->root->node_id);
-    } else {
+        return 1 if ($child->node_id != $child->root->node_id);
+        return 0;
+    }
+    return 0;
+}
+
+=head2 is_supertree
+
+  Example     : print "I'm a supertree" if $node->is_supertree();
+  Description : Detects and reports if a node is a supertree.
+  Returntype  : Boolean
+  Exceptions  :
```

Comment:
Not sure what to do with this line, but I feel if you leave it like it is, some people may interpret it as we have forgotten to add the information. Not sure if removing or adding `None` may be better options.